### DISCUSSION TITLE
Can this LoRaMac-node work on STM32F103C8T6 (blue pill)?

### DISCUSSION THREAD

**Discussion by shabaz-git** [Lora-net/LoRaMac-node#1055] [Q&A] [answered]
Hi,
I worked with Arduino Uno and RFM95w to create an end node h/w and the MCCI Arduino-lmic stack as s/w. As this stack supports only class A and B functionality, I want to use this (LoRaMac-node) stack on any stm32 MCU which meets the memory requirements as suggested by the Semtech(8Kb of SRAM and 12Kb of Flash). So, I found the _stm32 blue pill_ as an economical board meeting those memory requirements. But as most of the listed supported hardware consists of the **STM32L** series only. 
So, my queries are:

1. Can this stack be run on blue-pill? If not, what is the constraint? If yes, please provide steps to do.
2. If blue-pill won't work with this stack, is there any board that's having supported MCU other than listed in this(as they seem a bit expensive)?
3. If there is any chance of getting a Class C supported stack for Arduino, please provide the link.

**Comment by mluis1** [comment, marked as answer]
The STM32F103C8T6 doesn't look to be suitable to run this stack.

This MCU has 64KB FLASH memory. Compiling this project provided basic example in release mode gives already a memory usage bigger than 64KB. And your application will most certainly increase the required amount of memory.

Could you please provide the link for the Semtech documentation where it is said that the minimum MCU requirement is 12KB of FLASH?

Looking at following link it is said that minimal FLASH size is 128 KB FLASH memory.
https://lora-developers.semtech.com/library/tech-papers-and-guides/mcu-memory-management/

Also in following document it is said that the minimal FLASH size is 128 KB and the recommendation is 256 KB
[AN1200.28](https://semtech.my.salesforce.com/sfc/p/#E0000000JelG/a/2R000000HSRD/hyKkYJa7E.71emIBZ2LpgwFomMaTHRwCcb.dCogwdUg)

Concerning your other questions you will get I think better answers by using a general forum like
https://forum.lora-developers.semtech.com/

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Reply by shabaz-git** [reply]
Thanks for the response @mluis1 .
Regarding 12Kb of flash, it was a typing error. Yeah, the minimum suggested is 128kb. As few variants of blue pill(with STM32F103CB) are having 128kb of flash.
One more doubt was in src/boards/mcu/stm32/ section of this stack only STM32L* mcu hal are available. So, STM32F* can not be supported, right?

**Reply by mluis1** [reply]
All kind of MCUs can be supported in case it has the required features and a port has been implemented.
Unfortunately, we cannot provide support for all kind MCUs that exist in the World out of the box.

This means that to use STM32F* MCUs one must implement the porting. This porting should be quite straight forward as the ST HAL is quite similar for the different STM32 MCU families.

I'm not aware of an existing port for the STM32F* however maybe someone has already implemented such port.
Maybe you can try to search if such port exists or maybe someone aware of such port reading this can provide you the link for it.
